### PR TITLE
Adding jdk, findspark, and spark-2.3 to the repository

### PR DESCRIPTION
**Pull request by kmader**
Add the ability to run standard spark code in a notebook using findspark and the full spark installation

### CODE REVIEW

**Comment by rgbkrk** [comment]
Thanks! Based on locality of repo2docker builds, I'd prefer the mirror used be http://mirrors.ocf.berkeley.edu/apache/spark/spark-2.3.0/spark-2.3.0-bin-hadoop2.7.tgz (whereas for your local install please keep using your preferred mirror).

**Comment by rgbkrk** [comment]
What setup code should we provide in a notebook for this image?

**Comment by kmader** [comment]
Thanks, I made the changes, apparently the jdk9 has some issues with spark still. I've also added a 'preview mode' to look at the web-gui

File: postBuild [hunk 1]
```diff
@@ -1,2 +1,7 @@
-wget http://mirror.switch.ch/mirror/apache/dist/spark/spark-2.3.0/spark-2.3.0-bin-hadoop2.7.tgz
+wget http://mirrors.ocf.berkeley.edu/apache/spark/spark-2.3.0/spark-2.3.0-bin-hadoop2.7.tgz 
 tar -xzf spark-2.3.0-bin-hadoop2.7.tgz
+# set spark home and pyspark paths
+SPARK_HOME=$HOME/spark-2.3.0-bin-hadoop2.7
+export PATH=$SPARK_HOME/bin:$PATH
+export PYTHONPATH=$SPARK_HOME/python:$PYTHONPATH
+export PYSPARK_SUBMIT_ARGS="--master local[*] pyspark-shell"
```
Comment (rgbkrk) [inline postBuild:7]: Nice!

**Pull request merged**